src/lightning/pytorch/core/module.py

```diff
@@ -852,6 +885,18 @@ def predict_step(self, batch: Any, batch_idx: int, dataloader_idx: int = 0) -> A
         based accelerator. This happens for ``Trainer(strategy="ddp_spawn")``
         or training on 8 TPU cores with ``Trainer(accelerator="tpu", devices=8)`` as predictions won't be returned.
 
+        Args:
+            batch: The output of your :class:`~torch.utils.data.DataLoader`.
+            batch_idx: The index of this batch.
+            dataloader_idx: The index of the dataloader that produced this batch.
+                (only if multiple dataloaders used)
+
+        Return:
+            - :class:`~torch.Tensor` - The loss tensor
+            - ``dict`` - A dictionary. Can include any keys, but must include the key ``'loss'``.
+            - ``None`` - Skip to the next batch. This is only for automatic optimization.
+                This is not supported for multi-GPU, TPU, IPU, or DeepSpeed.
```

Comment:
Not sure what possible outputs to expect here. In TorchGeo, we'll likely want to return `dict[str, Tensor]`. So maybe `Optional[STEP_OUTPUT]` and just say "Predicted output"?

Reply:
Yep, could be anything